flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/api/window/CountWindow.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.api.window;
+
+import org.apache.flink.api.common.typeutils.SimpleTypeSerializerSnapshot;
+import org.apache.flink.api.common.typeutils.TypeSerializerSnapshot;
+import org.apache.flink.api.common.typeutils.base.TypeSerializerSingleton;
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.DataOutputView;
+import org.apache.flink.util.MathUtils;
+
+import java.io.IOException;
+
+/**
+ * A {@link Window} that represents a count window.
+ */
+public class CountWindow extends Window {
```

Comment:
Why named `CountWindow` instead of `GlobalWindow`?
DataStream use `GlobalWindow` + Trigger to implement count window, is that easier to understand?

Reply:
We didn't introduced `Evictor` to our window operator, so we cannot rely on the same mechanism to implement count window.

Reply:
Need some comments to illustrate the difference between `CountWindow` and DataStream's `GlobalWindow`?